test/e2e/network/network_policy.go

```diff
@@ -1744,3 +1747,69 @@ func cleanupNetworkPolicy(f *framework.Framework, policy *networkingv1.NetworkPo
 		framework.Failf("unable to cleanup policy %v: %v", policy.Name, err)
 	}
 }
+
+var _ = SIGDescribe("SCTP NetworkPolicy [Feature:SCTP] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("sctpnp")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		// SCTP in K8s is not supported on Windows
+		e2eskipper.SkipIfNodeOSDistroIs("windows")
+
+		if proxyMode, err := proxyMode(f); err == nil {
+			if proxyMode == "userspace" {
+				e2eskipper.Skipf("The test doesn't work with kube-proxy in userspace mode")
```

Comment:
I don't think you need to check this; while it's true that the userspace proxy doesn't support SCTP, that's why we have the tests behind a feature flag; if you're using the userspace proxy, then don't enable the `[Feature:SCTP]` tests.

Likewise for Windows; people running the tests on Windows should not enable `[Feature:SCTP]`. If Windows gets SCTP support later, they shouldn't need to go around modifying every test case to let them test it.

Given that, you don't need your own `BeforeEach` wrapper, so you can just move this in with the other NetworkPolicy tests...

Reply:
Fixed.